apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+    // Use a cursor, measures slightly faster than
+    try (ScrollableResults itemCursor =
+        em.unwrap(Session.class)
+            .createQuery(SELECT_UNHASHED_BENFICIARIES)
+            .setFetchSize(BATCH_COUNT)
+            .scroll(ScrollMode.SCROLL_INSENSITIVE)) {
+      int count = 0;
+      while (inPeriod(startTime, TASK_TIME_LIMIT) && itemCursor.next()) {
+        final Beneficiary beneficiary = (Beneficiary) itemCursor.get(0);
+        beneficiary
+            .getMedicareBeneficiaryId()
+            .ifPresent(
+                mbi -> {
+                  if (mbi.isEmpty()) return;
+                  final String mbiHash =
+                      "xxx"; // RifLoader.computeMbiHash(options, secretKeyFactory, mbi);
+                  beneficiary.setMbiHash(Optional.of(mbiHash));
+                });
+        isDone = itemCursor.isLast();
+        // Write to the DB in batches
+        if (++count % BATCH_COUNT == 0) {
+          em.flush();
+          em.clear();
+        }
+      }
+      beneficaryMeter.mark(count);
+      LOGGER.debug("Finished fixing up a Beneficiary batch: {}", count);
+    }
+    return isDone;
+  }
+
+  /**
+   * Fixup a batch of BeneficiaryHistory. Update the history metrics.
+   *
+   * @param em a {@link EntityManager} setup for a transaction
+   * @return true if done with all fixups
+   */
+  public Boolean fixupHistoryBatch(final EntityManager em, final Instant startTime) {
+    LOGGER.debug("Start fixing up a History batch");
+    boolean isDone = true;
+    try (ScrollableResults itemCursor =
+        em.unwrap(Session.class)
+            .createQuery(SELECT_UNHASHED_HISTORIES)
+            .setFetchSize(BATCH_COUNT)
+            .scroll(ScrollMode.FORWARD_ONLY)) {
+      int count = 0;
+      while (inPeriod(startTime, TASK_TIME_LIMIT) && itemCursor.next()) {
+        final BeneficiaryHistory beneficiary = (BeneficiaryHistory) itemCursor.get(0);
+        beneficiary
+            .getMedicareBeneficiaryId()
+            .ifPresent(
+                mbi -> {
+                  if (mbi.isEmpty()) return;
+                  final String mbiHash =
+                      "xxx"; // RifLoader.computeMbiHash(options, secretKeyFactory, mbi);
+                  beneficiary.setMbiHash(Optional.of(mbiHash));
+                });
+        isDone = itemCursor.isLast();
+        // Write to the DB in batches
+        if (++count % BATCH_COUNT == 0) {
+          em.flush();
+          em.clear();
+        }
+      }
+      historyMeter.mark(count);
+      LOGGER.debug("Finished fixing up a History batch: {}", count);
+    }
+    return isDone;
+  }
+
+  /**
+   * Any time left in this time slice?
+   *
+   * @param start of the period
+   * @param period duration
+   * @return true iff current period is less than the passed in period duration;
+   */
+  public static boolean inPeriod(final Instant start, final Duration period) {
+    return Duration.between(start, Instant.now()).compareTo(period) <= 0;
```

Comment:
I think this function would return `true` if `start` was after `now()`

Reply:
Agree. Will fix the function for this case.